tools/update_issues_with_description.php

```diff
@@ -0,0 +1,62 @@
+#!/usr/bin/php
+<?php
+/**
+ * The script update_issues_with_description.php copies the first comment
+ * by timestamp (dateAdded) in issues_comments table to the description
+ * field of the issues table matching based on the issueID
+ *
+ * Usage: php update_issues_with_description.php
+ *
+ * PHP Version 7
+ *
+ * @category Main
+ * @package  Loris
+ * @author   Loris Team <loris-dev@bic.mni.mcgill.ca>
+ * @license  Loris license
+ * @link     https://www.github.com/aces/Loris/
+ */
+
+require_once __DIR__ . "/generic_includes.php";
+
+set_include_path(
+    get_include_path().":".
+    __DIR__."/../project/libraries:".
+    __DIR__."/../php/libraries:"
+);
+
+require_once __DIR__ . "/../vendor/autoload.php";
+require_once "NDB_Client.class.inc";
+$client = new NDB_Client();
+$client->makeCommandLine();
+$client->initialize();
+
+$DB = \NDB_Factory::singleton()->database();
+
+$allIssueIDsWithComments = $DB->pselectCol(
+    "SELECT issueID FROM issues_comments",
+    []
+);
+
+foreach ($allIssueIDsWithComments as $issueID) {
+    $existingDescription = $DB->pselectOne(
+        "SELECT `description` FROM issues WHERE issueID=:issue_id",
+        ["issue_id" => $issueID]
+    );
+    if (!$existingDescription) {
+        $description = $DB->pselectOne(
+            "SELECT issueComment
+            FROM issues_comments ic
+            WHERE ic.issueID = :issue_id
+            ORDER BY ic.dateAdded ASC
+            LIMIT 1",
```

Comment:
pselectOne already adds a limit. Removing the "limit 1" here should fix your error.

Reply:
@driusan That gives another error, since the LIMIT is 2. Using pselect directly solves it. I am just doing some final tests, will commit after that. What are your thoughts?

Reply:
is the query expected to return more than 1 row? pselectOne does a LIMIT 2 because that's sufficient to efficiently tell if it returned zero rows and should be null, returned multiple rows and should throw an exception (because the code expected one value but the query matched many rows), or returned exactly 1 rows and it should be returned.

If the query is expected to have multiple matches and the code handles them properly then pselect is fine. But you should not run pselect with a "LIMIT 1" because that will just mask the root cause of the error and will return a random one of the multiple rows that the query matched.

Reply:
Yes without limit 1 it will return multiple rows.  The multiple matches will be handled by this `ORDER BY ic.dateAdded ASC LIMIT 1`. This will also ensure that there is no randomness. Shall I proceed with pselect?

Reply:
if there is an order by and you're sure the first one by date is the value that you want, then that would make sense.